Realm/RLMManagedDictionary.mm

```diff
@@ -277,17 +295,19 @@ - (void)removeObjectsForKeys:(NSArray *)keyArray {
 
 - (void)removeObjectForKey:(id)key {
     changeDictionary(self, ^{
-        RLMAccessorContext context(*_objectInfo);
+        RLMAccessorContext context(*_ownerInfo, *_objectInfo);
+        context.currentProperty = _property;
         _backingCollection.try_erase(context.unbox<realm::StringData>(key));
     });
 }
 
 - (void)enumerateKeysAndObjectsUsingBlock:(void (^)(id key, id obj, BOOL *stop))block {
-    RLMAccessorContext c(*_objectInfo);
+    RLMAccessorContext context(*_ownerInfo, *_objectInfo);
+    context.currentProperty = _property;
     BOOL stop = false;
     @autoreleasepool {
         for (auto&& [key, value] : _backingCollection) {
-            block(c.box(key), c.box(value), &stop);
+            block(context.box(key), [self objectForKey:context.box(key)], &stop);
```

Comment:
This is probably a significant pessimization to iteration performance?

Reply:
For nested collections we want to be able to return the nested collection, which makes use of the get_dictionary and get_list API